### DISCUSSION TITLE
how to define different mesh resolutions

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#21976] [Q&A General] [answered]
Hi

I need to apply to different mesh resolutions to the domain in the pictures. I believe moose has this already but I could not find an example.  Could someone give some direction?

![image](https://user-images.githubusercontent.com/88537645/187552414-d43db89e-d029-4fc9-989f-2045dc0bbfe3.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

You ll want to either use adaptivity or the RefineBlockGenerator.
https://mooseframework.inl.gov/source/meshgenerators/RefineBlockGenerator.html

Guillaume

**Reply by GiudGiud** [reply]
Adaptivity documentation is here 
https://mooseframework.inl.gov/syntax/Adaptivity/index.html
you ll want to set the marker uniformly by blocks

**Reply by aaelmeli** [reply]
> Hello
> 
> You ll want to either use adaptivity or the RefineBlockGenerator. https://mooseframework.inl.gov/source/meshgenerators/RefineBlockGenerator.html
> 
> Guillaume

Thanks, @GiudGiud , that helps.

**Reply by selarem** [reply]
Any example combining  SubdomainBoundingBoxGenerator and RefineBlockGenerator inside a GeneratedMeshGenerator.
It never worked for me !
regrads,

**Reply by GiudGiud** [reply]
need a MWE if something doesn't work
